merlin/dag/graph.py

```diff
@@ -30,8 +31,19 @@
 
 
 class Graph:
-    def __init__(self, output_node: Node):
+    def __init__(self, output_node: Node, subgraphs: Optional[Dict[str, Node]] = None):
         self.output_node = output_node
+        self.subgraphs = subgraphs or {}
+
+        parents_with_deps = self.output_node.parents_with_dependencies
+        for name, sg in self.subgraphs.items():
+            if sg not in parents_with_deps:
+                raise ValueError(f"Subgraph {name} does not exist in output_node.")
```

Comment:
Maybe something like "The output node of subgraph {name} ({self.subgraphs[name]}) does not exist among the parents or dependencies of {output_node}"?